### PR TITLE
Application upgrade

### PR DESCRIPTION
**Pull request by litvinets**
1. applied new API for application requests
2. added confirmation modals for reject/leave actions

related issue: https://github.com/ita-social-projects/OoS-Frontend/issues/1780

### CODE REVIEW

File: src/app/shell/personal-cabinet/parent/parent-applications/parent-applications.component.ts [hunk 1]
```diff
@@ -64,12 +72,22 @@ export class ParentApplicationsComponent extends CabinetDataComponent implements
...
   onLeave(application: Application): void {
-    const applicationUpdate = new ApplicationUpdate(application.id, Statuses.Left);
-    this.store.dispatch(new UpdateApplication(applicationUpdate));
+    const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: ModalConfirmationType.leaveWorkshop,
+      },
+    });
+    dialogRef.afterClosed().subscribe((result: string) => {
```
Comment (khrMykhailo) [inline src/app/shell/personal-cabinet/parent/parent-applications/parent-applications.component.ts:81]: unsubscribe pls

File: src/app/shell/personal-cabinet/provider/provider-applciations/provider-applciations.component.ts [hunk 2]
```diff
@@ -90,8 +99,15 @@ export class ProviderApplciationsComponent extends CabinetDataComponent implemen
    * @param Application event
    */
   onReject(application: Application): void {
-    const applicationUpdate = new ApplicationUpdate(application.id, Statuses.Rejected, application?.rejectionMessage);
-    this.store.dispatch(new UpdateApplication(applicationUpdate));
+    const dialogRef = this.matDialog.open(ReasonModalWindowComponent, {
+      data: { type: ModalConfirmationType.rejectApplication },
+    });
+    dialogRef.afterClosed().subscribe((result: string) => {
```
Comment (khrMykhailo) [inline src/app/shell/personal-cabinet/provider/provider-applciations/provider-applciations.component.ts:105]: unsubscribe pls

**Review by khrMykhailo — APPROVED** [review]

File: src/app/shell/personal-cabinet/parent/parent-applications/parent-applications.component.ts (hunk 1 above)
Comment (litvinets) [reply src/app/shell/personal-cabinet/parent/parent-applications/parent-applications.component.ts:81]: @khrMykhailo  You don't need to unsubscribe as the observable itself completes.You can verify the same by adding a finalize block to see whether observable completes itself or not.

File: src/app/shell/personal-cabinet/provider/provider-applciations/provider-applciations.component.ts (hunk 2 above)
Comment (litvinets) [reply src/app/shell/personal-cabinet/provider/provider-applciations/provider-applciations.component.ts:105]: @khrMykhailo  You don't need to unsubscribe as the observable itself completes.You can verify the same by adding a finalize block to see whether observable completes itself or not.

**Pull request merged**